src/compiler/transpile/test/transpile.spec.ts

```diff
@@ -155,4 +157,67 @@ describe('transpile', () => {
     expect(r.hasChangedJsText).toBe(false);
   });
 
+  it('should transpile with core and without typescript errors', async () => {
+    // this one takes a bit longer
+    jest.setTimeout(20 * 1000);
+
+    const nodeModulesDir = path.resolve(__dirname, '../../../../node_modules');
+    const distDir = path.resolve(__dirname, '../../../../dist')
+
+    c.config.suppressTypeScriptErrors = false;
+    c.config.buildAppCore = true;
+
+    // typescript needs real node modules and stencil dist files
+    // so we need to bypass the testing-fs for these.
+    const originalReadFileSync = c.fs.readFileSync.bind(c.fs);
+    const originalStatSync = c.fs.statSync.bind(c.fs);
+    c.fs.readFileSync = (filePath) => {
+      filePath = normalizePath(filePath);
+      if (filePath.indexOf(nodeModulesDir) === 0 || filePath.indexOf(distDir) === 0) {
+        return fs.readFileSync(filePath).toString();
+      }
+
+      return originalReadFileSync(filePath);
+    }
+    c.fs.statSync = (itemPath) => {
+      itemPath = normalizePath(itemPath);
+      if (itemPath.indexOf(nodeModulesDir) === 0 || itemPath.indexOf(distDir) === 0) {
+        return fs.statSync(itemPath);
+      }
+
+      return originalStatSync(itemPath);
+    }
```

Comment:
Not sure about the monkey-patching here. This could also be a feature of the `testing-fs` but that would've required more refactoring.